.github/workflows/terraform-observe_scheduler.yaml

```diff
@@ -49,8 +49,38 @@ jobs:
       - uses: hashicorp/setup-terraform@v2
         with:
           terraform_version: ${{ inputs.terraform-version }}
+
+      - name: Checkout Main
+        uses: actions/checkout@v3
+        if: startsWith(${{ matrix.directory }}, 'upgrade')
+        with:
+          ref: "main"
+          fetch-depth: '0'
+
+      - name: make test upgrade
+        if: startsWith(${{ matrix.directory }}, 'upgrade')
+        run: |
+
+          make test "${{ matrix.directory }} DO_NOT_RUN_DELETE"
+
+          git checkout ${{ github.head_ref }}
```

Comment:
These commands should each be broken out into their own steps for grouping of output in the UI. Move the `env` up to the job level to avoid repeating it.